public/components/overview/mitre/components/techniques/techniques.tsx

```diff
@@ -177,29 +189,73 @@ export const Techniques = withWindowSize(class Techniques extends Component {
     }
   }
 
+  async getMitreTechniques (params) {
+    this.setState({ isSearching: true })
+    return await WzRequest.apiReq("GET", "/mitre/techniques", { params: params })
+    .then((res) => res)
+    .catch((err) => {
+       this.showToast(
+         'danger',
+         'Error',
+         `Mitre techniques could not be fetched: ${err}`,
+         3000
+       );
+      []
+    });
+  }
+
+  async buildMitreTechniquesFromApi () {
+    const params = { limit: 500 };
+    const output = await this.getMitreTechniques(params)
+    const totalItems = (((output || {}).data || {}).data || {}).total_affected_items;
+    let mitreTechniques = [];
+    mitreTechniques.push(...output.data.data.affected_items);
+    if (totalItems && output.data && output.data.data && totalItems > 500) {
+      params.offset = 0;
+      while (mitreTechniques.length < totalItems && params.offset < totalItems) {
+        params.offset += params.limit;
+        const tmpData = await this.getMitreTechniques(params)
+        mitreTechniques.push(...tmpData.data.data.affected_items);
+      }
```

Comment:
if this is sync use `for` instead of `while` 
if this is async use `await Promise.all` 
using `while` with `await` inside may have bad behaviors

Reply:
Replaced to use with `Promise.all` and do the requests in parallel